### PR TITLE
feat: Add CommonJS and ESM build targets for Polaris Tokens

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR updates `@shopify/polaris-tokens` to export `CommonJS` and `ESM` build targets. This is accomplished with vanilla `Rollup` and `Babel` presets and Shopify’s `TypeScript` and `Browserslist` configs. Additionally, the post build transforms have been upgraded to TypeScript and ran with a zero config `ts-node` process in the new `build:assets` script.

#### Example `@shopify/polaris-tokens` exports
image

### CODE REVIEW

**Comment by alex-page** [comment]
@aaronccasanova I want to avoid all scope creep for tokens. I don't see these exports in https://github.com/Shopify/polaris-tokens/blob/main/package.json. Is this needed right now?

File: polaris-tokens/package.json [hunk 1]
```diff
@@ -2,13 +2,35 @@
   "name": "@shopify/polaris-tokens",
   "version": "5.0.0-rc.3",
   "description": "",
-  "main": "dist/index.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "browser": "dist/browser/index.js",
```
Comment (alex-page) [inline polaris-tokens/package.json]: Lets match this to previous tokens

**Review by alex-page — APPROVED** [review]

File: polaris-tokens/package.json (hunk 1 above)
Comment (aaronccasanova) [reply polaris-tokens/package.json]: Roger roger, removed the `browser` build target: https://github.com/Shopify/polaris/pull/5601/commits/78712ee13c7439a201dced7649dd9ca643f13a53

File: polaris-tokens/package.json [hunk 2]
```diff
@@ -2,13 +2,35 @@
...
+  "exports": {
+    ".": {
+      "types": "./dist/types/index.d.ts",
+      "import": "./dist/esm/index.js",
+      "require": "./dist/cjs/index.js"
+    }
+  },
   "scripts": {
-    "build": "tsc -b && node scripts",
+    "build": "yarn build:types && yarn build:js && yarn build:assets",
```
Comment (sam-b-rose) [inline polaris-tokens/package.json]: we have `npm-run-all` so could shorten this to `run-s build:*` if we wanted (I think)

File: polaris-tokens/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const extensions = ['.js', '.jsx', '.ts', '.tsx'];
+
+/**
+ * @type {import('rollup').RollupOptions}
+ */
+export default {
+  input: 'src/index.ts',
+  output: [
+    {
+      format: /** @type {const} */ ('cjs'),
```
Comment (sam-b-rose) [inline polaris-tokens/rollup.config.js:18]: Curious, what is the function of the `(...)` here?

**Review by sam-b-rose — APPROVED** [review]
This looks super clean! Love the use of babel + rollup to create all supported builds.

File: polaris-tokens/package.json (hunk 2 above)
Comment (aaronccasanova) [reply polaris-tokens/package.json]: Updated: https://github.com/Shopify/polaris/pull/5601/commits/f157771ee1abb683726b8ec370a9932e11702aa0

File: polaris-tokens/rollup.config.js (hunk 3 above)
Comment (aaronccasanova) [reply polaris-tokens/rollup.config.js:18]: This is fancy TS/JSDoc syntax to narrow `string` to the string literal `'cjs'`. The above `import('rollup').RollupOptions` expects a union of string literals.

Comment (aaronccasanova) [reply polaris-tokens/rollup.config.js:18]: Same as doing `format: 'cjs' as const,` in TypeScript
